lib/js_routes.rb

```diff
@@ -216,13 +216,19 @@ def build_js(route, parent_route)
   end
 
   def route_js_arguments(route, parent_spec)
+    parts = route.parts
     required_parts = route.required_parts.clone
-    optional_parts = route.parts - required_parts
-    default_parts = route.defaults.select do |part, _|
+    optional_parts = parts - required_parts
+    default_options = route.defaults.select do |part, _|
       FILTERED_DEFAULT_PARTS.exclude?(part) && URL_OPTIONS.include?(part) || required_parts.include?(part)
     end
     [
-      required_parts, optional_parts, serialize(route.path.spec, parent_spec), default_parts
+      parts,
+      required_parts,
+      optional_parts,
+      default_options,
+      Rails.version < '4.0',
+      serialize(route.path.spec, parent_spec)
```

Comment:
I like the hash idea. We can use `{"opt1": false,"arg1":true,"opt2":false,"arg2":true}` because `false` and `true` turn into `!0` and `!1` using JS minification. I think this is good enough.